src/app/pages/vm/vm-wizard/vm-wizard.component.ts

```diff
@@ -427,6 +454,26 @@ export class VMWizardComponent {
       const vcpu_limit = _.find(this.wizardConfig[1].fieldConfig, {'name' : 'vcpu_limit'});
       vcpu_limit.paraText = helptext.vcpus_warning + ` ${this.maxVCPUs} ` + helptext.vcpus_warning_b;
     })
+
+    this.ws.call("device.gpu_pci_ids_choices").subscribe((pci_choices: Object) => {
+      const isolated_gpu_pci_ids_conf = _.find(this.wizardConfig[5].fieldConfig, {name : "isolated_gpu_pci_ids"});
+      for(let key in pci_choices) {
+        isolated_gpu_pci_ids_conf.options.push({label: key, value: pci_choices[key]})
+      }
+    })
+
+    this.ws.call("device.get_gpus").subscribe((gpus) => {
+      this.gpus = gpus;
+      const gpu = _.find(this.wizardConfig[5].fieldConfig, {name : "gpu"});
+      for(let item of gpus) {
+        gpu.options.push({label: item.description, value: item.addr.pci_slot})
+      }
+    })
+
+    this.systemGeneralService.getAdvancedConfig.subscribe((res) => {
+      ( < FormGroup > this.entityWizard.formArray.get([5])).controls['isolated_gpu_pci_ids'].setValue(res.isolated_gpu_pci_ids)
```

Comment:
I am worried that if in the future someone makes changes to `formArray` and the element that you expect to be 5 becomes 6, it would not be easily noticeable. 
Is there anything else we can do here, i.e. query by some other condition and/or move to a separate getter?